### ISSUE TITLE
Should we allow plot_availability to deal with doy (day of year)?

### ISSUE THREAD

**Issue by j-harbin** [casaultb/azmpdata#65]
Hi All,

I'm curious your input on this. Currently for `plot_availability()`, when the user gives `areaType` to be `area`, the only `areaName` they have the option of giving is Gulf of St. Lawrence. The reasoning behind this is shown in the code below. Basically, Gulf of St. Lawrence is the only area that has "month" provided in addition to year, which is needed for the `plot_availability()`. Through further investigation, I see that some of the areas, for example, the remote sensing areas have "doy" aka day of year. I could add it into the code that if any area with "doy" rather than "month" could have the days converted to months and therefore be given as an argument for `plot_availability`?
``` r
library(azmpdata)
#> casaultb/azmpdata status:
#>  (Package ver: 0.2019.0.9100) Up to date
#>  (Data ver:2021-01-14 ) Up to date
k <- area_indexer(areaTypes = "area")
unique(k$area)
#>  [1] "scotian_shelf_box"    "E Georges Bank"       "Emerald Basin"       
#>  [4] "Georges Basin"        "Lurcher Shoal"        "Misaine Bank"        
#>  [7] "Cabot Strait"         "4Vn"                  "4Vs"                 
#> [10] "4W"                   "4X"                   "scotian_shelf_grid"  
#> [13] "4XeGoM+BoF"           "4XSS"                 NA                    
#> [16] "Gulf of St. Lawrence" "GSL+SS"               "CS_remote_sensing"   
#> [19] "CSS_remote_sensing"   "ESS_remote_sensing"   "GB_remote_sensing"   
#> [22] "LS_remote_sensing"    "WSS_remote_sensing"   "4V"                  
#> [25] "5Ze"
kk <- area_indexer(areaTypes="area", doMonths=T)
unique(kk$area)
#> [1] "Gulf of St. Lawrence" NA
```

<sup>Created on 2021-08-23 by the [reprex package](https://reprex.tidyverse.org) (v2.0.0)</sup>

**Comment by j-harbin** [comment]
Hi All.

I've added this to my pull request. The only dataset that this impacts is `RemoteSensing_Weekly_Broadscale`. This means that in addition to `plot_availability(areaType="area", areaName="Gulf of St. Lawrence", parameters="river_flux")`, the user also has the option to do `plot_availability(areaType="area", areaName="RemoteSensing_Weekly_Broadscale", parameters="surface_chlorophyll")` (see below). This is done by simply converting doy (day of year) into day, month, year.

``` r
library(azmpdata)
#> casaultb/azmpdata status:
#>  (Package ver: 0.2019.0.9100) Up to date
#>  (Data ver:2021-01-14 ) Up to date
data("RemoteSensing_Weekly_Broadscale")
plot_availability(areaType = "area", areaName="RemoteSensing_Weekly_Broadscale", parameters="surface_chlorophyll")
```

![](https://i.imgur.com/1uzAxTR.png)

<sup>Created on 2021-09-08 by the [reprex package](https://reprex.tidyverse.org) (v2.0.0)</sup>

**Comment by j-harbin** [comment]
Note this has been updated and is now considered done (and is in the pull request).

I made a mistake earlier on. When the user did `plot_availability(areaType="area")`, they had the option to give an areaName of Gulf of St. Lawrence or RemoteSensing_Weekly_Broadscale. The RemoteSensing_Weekly_Broadscale was a mistake on my end because that is the name of a dataset, not an area name. This is now fixed, and the user now has the option to give Gulf of St. Lawrence, or CS_remote_sensing, ESS_remote_sensing, CSS_remote_sensing, WSS_remote_sensing, GB_remote_sensing, LS_remote_sensing (which were all the area names within the RemoteSensing_Weekly_Broadscale dataset).
